catalog/core/catalog-core-standardframework/src/test/java/ddf/catalog/impl/operations/RemoteDeleteOperationsTest.java

```diff
@@ -0,0 +1,326 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p/>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p/>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+
+package ddf.catalog.impl.operations;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.mockito.Matchers.any;
+import static org.mockito.Matchers.isA;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+
+import javax.activation.MimeType;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import ddf.catalog.cache.solr.impl.ValidationQueryFactory;
+import ddf.catalog.content.impl.MockMemoryStorageProvider;
+import ddf.catalog.data.Metacard;
+import ddf.catalog.data.impl.MetacardImpl;
+import ddf.catalog.federation.FederationStrategy;
+import ddf.catalog.filter.proxy.adapter.GeotoolsFilterAdapterImpl;
+import ddf.catalog.filter.proxy.builder.GeotoolsFilterBuilder;
+import ddf.catalog.impl.FrameworkProperties;
+import ddf.catalog.impl.QueryResponsePostProcessor;
+import ddf.catalog.operation.DeleteRequest;
+import ddf.catalog.operation.DeleteResponse;
+import ddf.catalog.operation.impl.DeleteRequestImpl;
+import ddf.catalog.operation.impl.DeleteResponseImpl;
+import ddf.catalog.plugin.PostIngestPlugin;
+import ddf.catalog.plugin.PostResourcePlugin;
+import ddf.catalog.source.CatalogProvider;
+import ddf.catalog.source.CatalogStore;
+import ddf.catalog.source.IngestException;
+import ddf.catalog.source.Source;
+import ddf.catalog.source.SourceMonitor;
+import ddf.catalog.transform.InputTransformer;
+import ddf.catalog.util.impl.SourcePoller;
+import ddf.mime.MimeTypeResolver;
+import ddf.mime.MimeTypeToTransformerMapper;
+import ddf.mime.mapper.MimeTypeMapperImpl;
+
+public class RemoteDeleteOperationsTest {
+    DeleteOperations deleteOperations;
+
+    DeleteRequest deleteRequest;
+
+    DeleteResponse deleteResponse;
+
+    List<String> fanoutTagBlacklist;
+
+    RemoteDeleteOperations remoteDeleteOperations;
+
+    OperationsCatalogStoreSupport opsCatStoreSupport;
+
+    OperationsMetacardSupport opsMetacardSupport;
+
+    OperationsSecuritySupport opsSecuritySupport;
+
+    @Before
+    public void setUp() throws Exception {
```

Comment:
There's a lot going on in this function so it would probably be a good idea to break it into smaller, more descriptive helper functions.